### DISCUSSION TITLE
Gcloud compatible

### DISCUSSION THREAD

**Discussion by robiness** [jxstxn1/dockerize#108] [Ideas]
We need some preperation to be done before the actual docker image is build like 
 - creating the `.dockerize_build` and the `main.dart` inside it
 - build the actual app and move it in the `www` folder

I just can't find an easy solution to run the sidekick `build app` before using Cloud Build.
My workaround for now is to also git add both folders and always commit the changes with it.

I think [cloud builders](https://cloud.google.com/build/docs/cloud-builders) used in a `cloudbuild.yaml` are meant to do this kind of stuff but i didn't manage to get it working. 

I am thinking of playing around and building a cloud builder for us but only if you couldn't think of a easier solution to prepare everything before the actual docker image is built.

**Comment by robiness** [comment]
@passsy Your thoughts on this? 🤔

**Comment by jxstxn1** [comment]
We could try to setup Sidekick in a github action where we build the container and can then deploy it with the gcloud cli

**Comment by passsy** [comment]
I don't know what's the problem here. But I'd like to propose multiple solution to execute code before `cli build app`

## Multiple steps in cloud build 
Isn't cloud build like any other CI service where you can call bash scripts in any order?

```yaml
steps:
- name: 'prepare'
  args: ['cli', 'do', 'something']
- name: 'build'
  args: ['cli', 'build', 'app']
```

## Multiple steps in docker container
What about a [multi-stage docker file](https://docs.docker.com/build/building/multi-stage/#use-multi-stage-builds)?

## Adjust the build app command

The build app command is plain dart code. You can execute any code you want beforehand.
